software/deploy/deploy.go

```diff
@@ -289,3 +291,29 @@ func getAirflowUILink(deploymentID string, deploymentURLs []houston.DeploymentUR
 	}
 	return ""
 }
+
+func isDagOnlyDeploymentEnabled(appConfig *houston.AppConfig) bool {
+	return appConfig != nil && appConfig.Flags.DagOnlyDeployment
+}
+
+func isDagOnlyDeploymentEnabledForDeployment(deploymentInfo *houston.Deployment) bool {
+	return deploymentInfo != nil && deploymentInfo.DagDeployment.Type == houston.DagOnlyDeploymentType
+}
+
+func DagsOnlyDeploy(houstonClient houston.ClientInterface, appConfig *houston.AppConfig, deploymentID string) error {
+	// Throw error if the feature is disabled at Houston level
+	if !isDagOnlyDeploymentEnabled(appConfig) {
+		return ErrDagOnlyDeployDisabledInConfig
+	}
+
+	// Throw error if the feature is disabled at Deployment level
+	deploymentInfo, err := houston.Call(houstonClient.GetDeployment)(deploymentID)
+	if err != nil {
+		return fmt.Errorf("failed to get deployment info: %w", err)
+	}
+	if !isDagOnlyDeploymentEnabledForDeployment(deploymentInfo) {
+		return errDagOnlyDeployNotEnabledForDeployment
+	}
+
+	return nil
```

Comment:
I don't see any logic that would just deploy DAGs from the local `/dags` folder. Am I missing something? Could you add more description in the PR on what the expectations and behavior observed during testing?

Reply:
@neel-astro I will work on it now and will create a new PR for it.  Should I add a todo comment?
The idea is to create small PRs rather than a big one

Reply:
ok totally fine with that, could we then hide the flag from users, in case someone wants to cut out the release, they shouldn't be blocked on this 🙂

Reply:
cc: @kushalmalani  @sunkickr  Please review it, but don't merge it until the next PR which I will raise also gets approved. I will let you know.